### ISSUE TITLE
Miners connection fails

### ISSUE THREAD

**Issue by Nisefal** [sammy007/open-ethereum-pool#268]
#### System information

Geth version: 1.7.3-stable

Redis version: 3.0.7

OS & Version: Linux ( Ubuntu 16.04 )

#### Expected behaviour
1) geth is synced and running
2) open-ethereum-pool is running
3) on typing such ( EthDcrMiner64.exe -epool "mypoolip":8008 -ewal "myethereumwallet" -eworker "worker" -epsw x ) line in .bat file on mining computer and executing it mining process starts

#### Actual behaviour
1) geth is synced and running
2) open-ethereum-pool is running
3) what i see:

### Log from miner:

ETH: Stratum - connecting to '185.149.40.35' <185.149.40.35> port 8008
ETH: Stratum - Connected (185.149.40.35:8008)
ETH: Authorized
No pool specified for Decred! Ethereum-only mining mode is enabled
ETHEREUM-ONLY MINING MODE ENABLED (-mode 1)
Probably you are trying to mine Ethereum fork. Please specify "-allcoins 1" or "-allpools 1" option. Check "Readme" file for details.
**Pool sent wrong data, cannot set epoch**, disconnectETH: eth-proxy stratum mode **_(i think problem is in this line, but there is no info about it! )_**
Watchdog enabled
ETH: Connection lost, retry in 20 sec...  
Remote management (READ-ONLY MODE) is enabled on port 3333

ETH: Connection lost, retry in 20 sec... 
GPU0 t=0C fan=0%, GPU1 t=0C fan=0%, GPU2 t=0C fan=0%, GPU3 t=0C fan=0%, GPU4 t=0C fan=0%, GPU5 t=0C fan=0%, GPU6 t=0C fan=0%, GPU7 t=0C fan=0%
ETH: Stratum - connecting to '185.149.40.35' <185.149.40.35> port 8008
ETH: Stratum - Connected (185.149.40.35:8008)
ETH: Authorized
GPU0 t=0C fan=0%, GPU1 t=0C fan=0%, GPU2 t=0C fan=0%, GPU3 t=0C fan=0%, GPU4 t=0C fan=0%, GPU5 t=0C fan=0%, GPU6 t=0C fan=0%, GPU7 t=0C fan=0%
ETH: Job timeout, disconnect, retry in 20 sec... 
ETH: Stratum - connecting to '185.149.40.35' <185.149.40.35> port 8008
ETH: Stratum - Connected (185.149.40.35:8008)
ETH: Authorized
GPU0 t=0C fan=0%, GPU1 t=0C fan=0%, GPU2 t=0C fan=0%, GPU3 t=0C fan=0%, GPU4 t=0C fan=0%, GPU5 t=0C fan=0%, GPU6 t=0C fan=0%, GPU7 t=0C fan=0%
ETH: Job timeout, disconnect, retry in 20 sec...

and so on

### Pool  log
2017/12/29 01:26:22 Loading config: /home/pooladminus/open-ethereum-pool/config.json
2017/12/29 01:26:22 Running with 2 threads
2017/12/29 01:26:22 Backend check reply: PONG
2017/12/29 01:26:22 Starting API on 0.0.0.0:8080
2017/12/29 01:26:22 Set stats collect interval to 5s
2017/12/29 01:26:22 Set purge interval to 10m0s
2017/12/29 01:26:22 Policy state refresh complete
2017/12/29 01:26:22 Set policy stats reset every 1h0m0s
2017/12/29 01:26:22 Set policy state refresh every 1m0s
2017/12/29 01:26:22 Running with 8 policy workers
2017/12/29 01:26:22 Upstream: main => http://127.0.0.1:8545
2017/12/29 01:26:22 Upstream: backup => http://127.0.0.2:8545
2017/12/29 01:26:22 Default upstream: main => http://127.0.0.1:8545
2017/12/29 01:26:22 Stratum listening on 0.0.0.0:8008
2017/12/29 01:26:22 Purged stale stats from backend, 0 shares affected, elapsed time 50.065033ms
2017/12/29 01:26:22 Stats collection finished 482.76µs
2017/12/29 01:26:22 New block to mine on main at height 1 / 0x5241e1cc
2017/12/29 01:26:22 Set block refresh every 120ms
2017/12/29 01:26:22 Starting proxy on 0.0.0.0:8888
2017/12/29 01:26:22 Broadcasting new job to 0 stratum miners
2017/12/29 01:26:22 Jobs broadcast finished 22.68µs
2017/12/29 01:26:27 Stats collection finished 1.00254ms
2017/12/29 01:26:32 Stats collection finished 1.016881ms
2017/12/29 01:26:37 Stats collection finished 2.115205ms
2017/12/29 01:26:40 Stratum miner connected 0x63ad2840c6d1388da60acce70147a0dffdd12121@93.72.79.123
2017/12/29 01:26:42 Stats collection finished 740.988µs
2017/12/29 01:26:47 Stats collection finished 866.064µs
2017/12/29 01:26:52 Stats collection finished 903.756µs
2017/12/29 01:26:57 Stats collection finished 566.28µs
2017/12/29 01:27:02 Stats collection finished 633.048µs
2017/12/29 01:27:05 Client 93.72.79.123 disconnected
...
sometimes it randomly posts this line
**2017/12/29 01:32:48 Error reading from socket: read tcp 185.149.40.35:8008->93.72.79.123:50090: read: connection reset by peer**

#### Steps to reproduce the behaviour

1. install open-ethereum-pool step-by-step according the guide (done)
2. find a computer with Claymore miner (done)
3. try to connect to pool (done)
4. get in trouble (done) :D 

#### What i've already tried
using my pool's ip or 0.0.0.0 in config.json lead to this result
connecting to another open-ethereum-pool with similar ( EthDcrMiner64.exe -epool "mypoolip":8008 -ewal "myethereumwallet" -eworker "worker" -epsw x ) line is successful
i was trying to connect to 8888, 8080 (i know it couldn't work, but you try everything after three or so days of attempts to find any info about one line of text)
re-re-recheking config.json file and validating it according the guide
re-re-reinstalling pool/go-ethereum
dropping and re-redownloading chaindata with geth

#### What I expect:

_**IT MAGICALLY STARTS WORKING!**_ (no)
I'd be happy to find information why I have error with reading socket or where to look first. No way I want somebody to make it work. I'm going to administrate this pool in future and having no idea how does it works is not good. If somebody could provide documentation or link on documentation or similar topics or no-matter-what related to my problem, i'd be really happy. Thanks.
````
[backtrace]
````

**Comment by ahrcoin** [comment]
Did you try setting the -allcoins 1 option in claymore? I am using musicoin and must set it because of differing epoch sizes.

**Comment by Nisefal** [comment]
Yes, but there was no result, only message "use it with with -allcoin 1" dissapeared.

**Comment by ahrcoin** [comment]
Hmmmmm I have not had any issues with mine.  My miner connects with no problem.  I do get some sluggish share accepted messages every once in a while but have not have any connection issues.  I had issues before with like a broccoli plugin error that i could not get past and i ended up having to flash my vps and start over with fresh install of Ubuntu.  Musicmine.us  is what I finally got working tonight.  I can send you copies of any of my code if you think it would help.  I am not an expert by any means at this point but am still trying to learn.

**Comment by Zumili** [comment]
Nisefal did you have any further success to solve your problem? Because I just run into the same problem and also stuck their now for some while.

**Comment by cosmoarunn** [comment]
Use ethdcrminer64
What's your command line options?

**Comment by cosmoarunn** [comment]
i used ethdcrminer64
miners connected but the error was
"No new jobs were received for 10 mins - something wrong with the pool ... "
anyone familier with this error?

**Comment by ahrcoin** [comment]
Make sure your geth is still running and synced. You will run into this if geth has stopped for some reason.

**Comment by GRinvest** [comment]
This problem occurs when the network does not find a long block, for example coin dbix. Tell me please where to add work in the pool so that you can maintain long blocks?

**Comment by idkfakata** [comment]
#ETH miners: Old Claymore miner versions do not support mining after DAG epoch 200 which just passed with #Ethereum block 6000000 (error: Setting epoch failed). In order to resolve this issue please upgrade to the latest Claymore miner version.
